zeroconf/__init__.py

```diff
@@ -2831,34 +2920,26 @@ def handle_response(self, msg: DNSIncoming) -> None:
         if not updates and not address_adds and not other_adds and not removes:
             return
 
-        # Only hold the lock if we have updates
-        with self._handlers_lock:
-            for record in updates:
-                self.update_record(now, record)
-            # The cache adds must be processed AFTER we trigger
-            # the updates since we compare existing data
-            # with the new data and updating the cache
-            # ahead of update_record will cause listeners
-            # to miss changes
-            #
-            # We must process address adds before non-addresses
-            # otherwise a fetch of ServiceInfo may miss an address
-            # because it thinks the cache is complete
-            #
-            # The cache is processed under the lock to ensure
-            # that any ServiceBrowser that is going to call
-            # zc.get_service_info will see the cached value
-            # but ONLY after all the record updates have been
-            # processsed.
-            for record in address_adds:
-                self.cache.add(record)
-            for record in other_adds:
-                self.cache.add(record)
-            # Removes are processed last since
-            # ServiceInfo could generate an un-needed query
-            # because the data was not yet populated.
-            for record in removes:
-                self.cache.remove(record)
+        self.update_records(now, updates)
```

Comment:
If we used this in more places it would probably make sense to be a context manager

Reply:
If I don't make this a context manager its more likely to break in refactoring later so lets do that.